### PR TITLE
feat(route): add oreno3d.com - 俺の3Dエロ動画

### PR DESCRIPTION
**Pull request by xueli-sherryli**
## 该 PR 相关 Issue / Involved issue

Close #

## 完整路由地址 / Example for the proposed route(s)

```routes
/oreno3d/search/bronya/latest
/oreno3d/characters/283/latest
/oreno3d/authors/3189/latest
/oreno3d/tags/177/latest
/oreno3d/origins/3/latest
```

## 新 RSS 检查列表 / New RSS Script Checklist
  
- [x] 新的路由 New Route
  - [x] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [x] 文档说明 Documentation
  - [x] 中文文档 CN
  - [x] 英文文档 EN
- [ ] 全文获取 fulltext
  - [ ] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [ ] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [ ] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

## 说明 / Note
oreno3d.com 是 iwara.tv的第三方搜索引擎,可弥补iwara.tv自身搜索引擎的不足

### CODE REVIEW

File: docs/anime.md [hunk 1]
```diff
@@ -600,3 +600,53 @@ Sources
...
 
 <Route author="junfengP" example="/zdfx" path="/zdfx"/>
+
+## oreno3d - 俺の3Dエロ動画
+::: tip 提示
+可配合其他RSS解析库(如Python的feedparser库)实现视频的更新检测以及自动下载
+
+:::
+### 关键词搜索
+<Route author="xueli-sherryli" example="/oreno3d/search/latest/布洛妮娅" path="/oreno3d/search/:sort?/:keyword" :paramsDesc="['排序方法,见下表,默认为latest', '关键词']">
```
Comment (TonyRL) [inline docs/anime.md:610]: Please add linebreak and use the right punctuation
```suggestion
### 关键词搜索

<Route author="xueli-sherryli" example="/oreno3d/search/latest/布洛妮娅" path="/oreno3d/search/:sort?/:keyword" :paramsDesc="['排序方法，见下表，默认为 `latest`', '关键词']">
```

File: docs/anime.md [hunk 2]
```diff
@@ -600,3 +600,53 @@ Sources
...
+</Route>
+
+### 原作搜索
+<Route author="xueli-sherryli" example="/oreno3d/origins/latest/3" path="/oreno3d/origins/:sort?/:originid" :paramsDesc="['排序方法,见下表,默认为latest', '原作id,可在浏览器地址栏中找到']">
+
+| favorites | hot    | latest | popularity |
+| --------- | ------ | ------ | ---------- |
+| 高評価    | 急上昇 | 新着   | 人気       |
+
+</Route>
```
Comment (TonyRL) [inline docs/anime.md:652]: Please add en docs.

File: lib/v2/oreno3d/authors.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
+const got = require('@/utils/got');
+const { art } = require('@/utils/render');
+const path = require('path');
+const cheerio = require('cheerio');
+const get_sec_page_data = require("./get_sec_page_data");
+
+const rootUrl = 'https://oreno3d.com';
+
+const sortrename = {
```
Comment (TonyRL) [inline lib/v2/oreno3d/authors.js:9]: camelCase
```suggestion
const sortRename = {
```

File: lib/v2/oreno3d/authors.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    popularity: '人気',
+};
+
+const selector = 'a.box';
+module.exports = async (ctx) => {
+    // 下载
+    const sort = ctx.params.sort ?? 'latest';
+    const authorid = ctx.params.authorid ;
+    const userUrl = `${rootUrl}/authors/${authorid}?sort=${sort}`;
+    const response = await got.get(userUrl);
```
Comment (TonyRL) [inline lib/v2/oreno3d/authors.js:22]:
```suggestion
    const response = await got(userUrl);
```

File: lib/v2/oreno3d/authors.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+            const characters = sec_data.characters;
+            const tags = sec_data.tags;
+            const desc = sec_data.desc;
+            const iwara_link = sec_data.iwara_link;
+            const oreno3d_link = sec_data.oreno3d_link;
+            // 打包
+            return {
+                title: `${video_name} - ${authors}`,
+                author:authors,
+                link:oreno3d_link,
```
Comment (TonyRL) [inline lib/v2/oreno3d/authors.js:48]: You can put tags into `category`
```suggestion
                    link: oreno3d_link,
                    category: tags.split(' '),
```

File: lib/v2/oreno3d/get_sec_page_data.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    const raw_pic_selector = "header > figure.video-figure > a ";
+    const video_name_selector = "header > h1.video-h1 ";
+    const author_selector = "section:nth-child(4) > a > div.video-center";
+    const origins_selector = "section:nth-child(5) > a > div.video-center";
+    const characters_selector = "section:nth-child(6) > a > div.video-center";
+    const tags_selector = "section:nth-child(7) > ul > li > a div.tag-text";// 未修复
+    const desc_selector = "section blockquote.video-information-comment";
+    const iwara_link_selector = "header > figure.video-figure > a";
+    // 下载
+    const response = await got.get(link);
```
Comment (TonyRL) [inline lib/v2/oreno3d/get_sec_page_data.js:19]:
```suggestion
    const response = await got(link);
```

File: lib/v2/oreno3d/maintainer.js [hunk 7]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+    '/search/:sort?/:keyword': ['xueli_sherryli'],
+    '/characters/:sort?/:characterid': ['xueli_sherryli'],
+    '/authors/:sort?/:authorid': ['xueli_sherryli'],
+    '/tags/:sort?/:tagid': ['xueli_sherryli'],
+    '/origins/:sort?/:originid': ['xueli_sherryli'],
```
Comment (TonyRL) [inline lib/v2/oreno3d/maintainer.js:6]: Please sort route paths in alphabetical order
```suggestion
    '/authors/:sort?/:authorid': ['xueli_sherryli'],
    '/characters/:sort?/:characterid': ['xueli_sherryli'],
    '/origins/:sort?/:originid': ['xueli_sherryli'],
    '/search/:sort?/:keyword': ['xueli_sherryli'],
    '/tags/:sort?/:tagid': ['xueli_sherryli'],
```

File: lib/v2/oreno3d/radar.js [hunk 8]
```diff
@@ -0,0 +1,37 @@
+module.exports = {
+    'oreno3d.com': {
+        _name: 'oreno3d',
+        ecchi: [
```
Comment (TonyRL) [inline lib/v2/oreno3d/radar.js:4]: `ecchi.oreno3d.com` is 404 for me.
So I guess it should be
```suggestion
        '.': [
```

File: lib/v2/oreno3d/router.js [hunk 9]
```diff
@@ -0,0 +1,7 @@
+module.exports = function (router) {
+    router.get('/search/:sort?/:keyword', require('./search'));
+    router.get('/characters/:sort?/:characterid', require('./characters'));
+    router.get('/authors/:sort?/:authorid', require('./authors'));
+    router.get('/tags/:sort?/:tagid', require('./tags'));
+    router.get('/origins/:sort?/:originid', require('./origins'));
```
Comment (TonyRL) [inline lib/v2/oreno3d/router.js:6]: Please sort route paths in alphabetical order
```suggestion
    router.get('/authors/:sort?/:authorid', require('./authors'));
    router.get('/characters/:sort?/:characterid', require('./characters'));
    router.get('/origins/:sort?/:originid', require('./origins'));
    router.get('/search/:sort?/:keyword', require('./search'));
    router.get('/tags/:sort?/:tagid', require('./tags'));
```

File: lib/v2/oreno3d/templates/description.art [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+<p>作者：<b>{{ authors }}</b></p>
+<p>原作：<b>{{ origins }}</b></p>
+<p>角色：<b>{{ characters }}</b></p>
+<p>标签：<b>{{ tags }}</b></p>
+<p>简介：</p>
+<div style = "white-space: pre-line;">
+{{ desc }}
+</div>
+<p>iwara链接：<b>{{ iwara_link }}</b></p>
+<p>Oreno3D链接：<b>{{ oreno3d_link }}</b></p>
```
Comment (TonyRL) [inline lib/v2/oreno3d/templates/description.art:12]: Consider make it clickable instead of plaintext

**Review by TonyRL — COMMENTED** [review]
Please remove changes on `yarn.lock` unless you add a new package.
Please put optional parameters(`sort` in this case) to the end of route paths

File: docs/anime.md (hunk 2 above)
Comment (xueli-sherryli) [reply docs/anime.md:656]: added

File: lib/v2/oreno3d/characters.js [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+    popularity: '人気',
+};
+
+const selector = 'a.box';
+module.exports = async (ctx) => {
+    // 下载
+    const sort = ctx.params.sort ?? 'latest';
+    const characterid = ctx.params.characterid ;
+    const userUrl = `${rootUrl}/characters/${characterid}?sort=${sort}`;
+    const response = await got.get(userUrl);
```
Comment (TonyRL) [inline lib/v2/oreno3d/characters.js:22]:
```suggestion
    const response = await got(userUrl);
```
https://github.com/DIYgod/RSSHub/pull/10458#discussion_r945180004

File: lib/v2/oreno3d/origins.js [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+    popularity: '人気',
+};
+
+const selector = 'a.box';
+module.exports = async (ctx) => {
+    // 下载
+    const sort = ctx.params.sort ?? 'latest';
+    const originid = ctx.params.originid ;
+    const userUrl = `${rootUrl}/origins/${originid}?sort=${sort}`;
+    const response = await got.get(userUrl);
```
Comment (TonyRL) [inline lib/v2/oreno3d/origins.js:22]:
```suggestion
    const response = await got(userUrl);
```

File: lib/v2/oreno3d/search.js [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+    popularity: '人気',
+};
+
+const selector = 'a.box';
+module.exports = async (ctx) => {
+    // 下载
+    const sort = ctx.params.sort ?? 'latest';
+    const keyword = ctx.params.keyword ;
+    const userUrl = `${rootUrl}/search?sort=${sort}&keyword=${keyword}`;
+    const response = await got.get(userUrl);
```
Comment (TonyRL) [inline lib/v2/oreno3d/search.js:22]:
```suggestion
    const response = await got(userUrl);
```

File: lib/v2/oreno3d/tags.js [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+    popularity: '人気',
+};
+
+const selector = 'a.box';
+module.exports = async (ctx) => {
+    // 下载
+    const sort = ctx.params.sort ?? 'latest';
+    const tagid = ctx.params.tagid ;
+    const userUrl = `${rootUrl}/tags/${tagid}?sort=${sort}`;
+    const response = await got.get(userUrl);
```
Comment (TonyRL) [inline lib/v2/oreno3d/tags.js:22]:
```suggestion
    const response = await got(userUrl);
```

File: docs/en/anime.md [hunk 15]
```diff
@@ -105,3 +105,61 @@ For example: <https://www.webtoons.com/zh-hant/drama/gongzhuweimian/list?title_n
 ### [Naver](https://comic.naver.com)
 
 <RouteEn author="zfanta" example="/webtoons/naver/651673" path="/webtoons/naver/:titleId" :paramsDesc="['titleId of naver webtoon']" />
+
+## My 3D Hentai Videos(oreno3d)
```
Comment (TonyRL) [inline docs/en/anime.md:109]:
```suggestion
## 俺の3Dエロ動画(oreno3d)
```

**Review by TonyRL — COMMENTED** [review]
Since all 5 routes are almost the same, consider combining them into one single file.

**Pull request merged**